LanguageExt.Core/Prelude_Union.cs

```diff
@@ -0,0 +1,366 @@
+﻿namespace LanguageExt
+{
+	using System;
+
+	public static partial class Prelude
+	{
+		public static IUnion<B> Bind<A, B>(this IUnion<A> a, Func<A, IUnion<B>> func) =>
```

Comment:
I'll clean that up.  I've adapted this from a personal project that was more or less just to see if I could so I'm actually really happy to fix any issues you spot.   I'll fix that naming issue and I've got some more unit tests to add as well.  this will be in my next check-in.